HBASE-28211 BucketCache.blocksByHFile may leak on allocationFailure or if we reach io errors tolerated

wchevreuil opened the pull request.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m  2s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 11s |  master passed  |
| +1 :green_heart: |  compile  |   2m 37s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 37s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 43s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 30s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 35s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 35s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  10m 45s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 36s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m  9s |  The patch does not generate ASF License warnings.  |
|  |   |  36m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux bb547a59bc3e 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9631af791e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 12s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 44s |  master passed  |
| +1 :green_heart: |  compile  |   1m  0s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 17s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 58s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 222m 57s |  hbase-server in the patch passed.  |
|  |   | 247m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 4bab742fed9c 5.4.0-166-generic #183-Ubuntu SMP Mon Oct 2 11:28:33 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9631af791e |
| Default Java | Eclipse Adoptium-11.0.17+8 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/testReport/ |
| Max. process+thread count | 5048 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 43s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 43s |  master passed  |
| +1 :green_heart: |  compile  |   1m  7s |  master passed  |
| +1 :green_heart: |  shadedjars  |   6m 43s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 33s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 18s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 56s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 271m 56s |  hbase-server in the patch failed.  |
|  |   | 301m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux fca691c6586f 5.4.0-163-generic #180-Ubuntu SMP Tue Sep 5 13:21:23 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9631af791e |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/testReport/ |
| Max. process+thread count | 4235 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/1/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 56s |  master passed  |
| +1 :green_heart: |  compile  |   2m 51s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 56s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 26s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 59s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 47s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 48s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m  6s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  41m  4s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux b406c9efcee8 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e88daed9fb |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 82 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache9 commented on hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java:
```diff
@@ -642,12 +643,14 @@ void blockEvicted(BlockCacheKey cacheKey, BucketEntry bucketEntry, boolean decre
+      }
     }
     if (evictedByEvictionProcess) {
       cacheStats.evicted(bucketEntry.getCachedTime(), cacheKey.isPrimary());
```
So we could throw exceptions here?

Apache9 replied:
What about this one? Why we need to move this up?

Apache9 replied:
Ping @wchevreuil

wchevreuil replied:
Sorry, missed the previous comment. 

It was wrong to call the removeFileFromPrefetch if we are not decrementing the block number. In some occasions, we do replace the same block and call blockEvicted with decrementBlockNumber set to false. This means even though we evicted, we still have that block for the file and we shouldn't remove it from the list of fully cached files.

Apache9 commented on hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java:
```diff
@@ -221,13 +221,14 @@ public class BucketCache implements BlockCache, HeapSize {
-    }
-    return Long.compare(a.getOffset(), b.getOffset());
-  });
+  protected final NavigableSet<BlockCacheKey> blocksByHFile =
```
Just change it to package private if we want to access it in test

Apache9 reviewed (commented):
To be honest I can not fully understand why the description on jira will lead to code change in this PR... Especially you mentioned we may still fail to cache the entry so the entry in blocksByHFile will be there forever, but then we moved 'blocksByHFile.add' to the very beginning? Why?

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 12s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 18s |  master passed  |
| +1 :green_heart: |  compile  |   0m 51s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 28s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 223m  5s |  hbase-server in the patch failed.  |
|  |   | 246m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux f7deccaeafab 5.4.0-166-generic #183-Ubuntu SMP Mon Oct 2 11:28:33 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e88daed9fb |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/testReport/ |
| Max. process+thread count | 4588 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 21s |  master passed  |
| +1 :green_heart: |  compile  |   0m 40s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 45s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 43s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 230m 20s |  hbase-server in the patch failed.  |
|  |   | 251m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 7c929bedbad5 5.4.0-163-generic #180-Ubuntu SMP Tue Sep 5 13:21:23 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e88daed9fb |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/testReport/ |
| Max. process+thread count | 4251 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

wchevreuil commented:
> To be honest I can not fully understand why the description on jira will lead to code change in this PR... Especially you mentioned we may still fail to cache the entry so the entry in blocksByHFile will be there forever, 

If you look at the original code, we were adding the block [here](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java#L1167), which is before we have actually cached the block. If you end up getting a BucketAllocatorException, we then never remove the block from blocksByHFile.

> but then we moved 'blocksByHFile.add' to the very beginning? Why?

We moved it to the `putIntoBackingMap` method, [which is only called after we made sure the block was cached](https://github.com/wchevreuil/hbase/blob/HBASE-28211/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java#L1231). That way we don't end up with a `blocksByHFile` that's inconsistent with the `backingMap` and the cache itself.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 12s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 51s |  master passed  |
| +1 :green_heart: |  compile  |   0m 47s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 48s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 42s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 48s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 54s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 222m 48s |  hbase-server in the patch failed.  |
|  |   | 245m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ad547a3a15c6 5.4.0-166-generic #183-Ubuntu SMP Mon Oct 2 11:28:33 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e88daed9fb |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/testReport/ |
| Max. process+thread count | 4474 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 19s |  master passed  |
| +1 :green_heart: |  compile  |   0m 41s |  master passed  |
| +1 :green_heart: |  shadedjars  |   4m 42s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 236m 46s |  hbase-server in the patch failed.  |
|  |   | 258m  8s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux a025054c4908 5.4.0-163-generic #180-Ubuntu SMP Tue Sep 5 13:21:23 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e88daed9fb |
| Default Java | Temurin-1.8.0_352-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/testReport/ |
| Max. process+thread count | 4255 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/3/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  3s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/4/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  2s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/4/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  3s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/5/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  3s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/5/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  2s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/6/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  3s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/6/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  4s |  https://github.com/apache/hbase/pull/5530 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/6/console |
| versions | git=2.25.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 34s |  master passed  |
| +1 :green_heart: |  compile  |   3m  1s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 45s |  master passed  |
| +1 :green_heart: |  spotless  |   0m 55s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 36s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 57s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  13m  7s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.6.  |
| +1 :green_heart: |  spotless  |   0m 52s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m  7s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 17s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 6103e96c6cc1 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1203c2014b |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 77 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 14s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 18s |  master passed  |
| +1 :green_heart: |  compile  |   0m 54s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 26s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 28s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  3s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 263m 40s |  hbase-server in the patch failed.  |
|  |   | 288m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5530 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 814d075fd5a8 5.4.0-166-generic #183-Ubuntu SMP Mon Oct 2 11:28:33 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 1203c2014b |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/testReport/ |
| Max. process+thread count | 4730 (vs. ulimit of 30000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5530/7/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.
